packages/preferences/src/node/json-preference-server.ts

```diff
@@ -99,17 +106,29 @@ export class JsonPreferenceServer implements PreferenceServer {
 
     protected doReconcilePreferences(preferences: any | undefined) {
         if (preferences) {
-            if (this.preferences) {
-                this.fireChanged(this.preferences, preferences);
+            const validatedPreferences = this.validatePreferences(preferences);
+            if (validatedPreferences) {
+                this.fireChanged(this.preferences, validatedPreferences);
             } else {
-                this.fireNew(preferences);
+                this.fireNew(validatedPreferences);
             }
         } else if (this.preferences) {
             this.fireRemoved(this.preferences);
         }
         this.preferences = preferences;
     }
 
+    protected validatePreferences(preferences: any): any {
+        const validatedPrefs: any = {};
+        for (const preferenceName in preferences) {
+            const value = preferences[preferenceName];
+            if (ajv().validate(this.schema, { preferenceName: value })) {
```

Comment:
you can also do:
```ts
const ajv = new Ajv();
if (ajv.validate(this.schema, preferences)) {
    return preferences;
}
const result = {};
for (const name in preferences) {
    if (name is valid based on `ajv.errors`) { //https://github.com/epoberezkin/ajv#validation-errors
        result[name] = preferences[name];
    }
}
return result;
```

instead of recompiling the schema over and over, you even can create `Ajv` object in the constructor and compile the schema only once using for validation `ajv.validate(preferences)`